### PR TITLE
New build system

### PR DESCRIPTION
**Pull request by ldez**
- add dep
- replace `build.sh` by `Makefile`
- use multi-stage build

Closes #5

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,12 @@
...
+RUN make dependencies
+RUN make build
+
 # Create a minimal container to run a Golang static binary
 FROM scratch
-COPY whoami /
+COPY --from=builder /go/src/github.com/containous/whoami/whoami .
 ENTRYPOINT ["/whoami"]
-EXPOSE 80
+EXPOSE 80
```
Comment (dduportal) [inline Dockerfile]: Sorry to nitpick here but as http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_206 , a new line is required at the end of the file.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,11 @@
+
+default: build
```
Comment (dduportal) [inline Makefile]: So a `.PHONY` target should be added, to express the fact that Make should not have to search for the files `default`, `build`, `dependencies` or `image`.

The target can be put wherever you want it and shall look like this:

```
.PHONY: default build dependencies image
```

Ref. https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html

**Review by dduportal — APPROVED** [review]
LGTM

**Review by emilevauge — APPROVED** [review]
LGTM

**Review by mmatur — APPROVED** [review]
LGTM

**Comment by proudobinna9-source** [comment]
🚀 **ZeebCloud preview deployed** for this PR: http://pr-9-preview-close.localhost

**Pull request merged**